### ISSUE TITLE
[Epic] Sliding Sync in Element Web

### ISSUE THREAD

**Issue by novocaine** [element-hq/element-web#22264]
## Why?

Sync performance is poor, both on initial and incremental syncs, and scales linearly with the number of joined rooms.

## What?

We’ll integrate support for [Sliding Sync](https://github.com/matrix-org/matrix-doc/pull/3575) which provides a sync mechanism that scales independently of the number of rooms, enabling higher performance for all syncs. 

## Plan

- [x] Initial spike: https://github.com/matrix-org/matrix-react-sdk/pull/8328
- [x] Implement support for [MSC3575](https://github.com/matrix-org/matrix-doc/pull/3575) in Element Web
- [ ] Read proxy URL from `.well-known/matrix/client`, enabling SS by default if some `default: true` flag is set on the response, otherwise, keep it toggleable in labs
- [ ] Deploy the proxy to `:element.io` which then advertises the proxy so Element people can play with it very easily

**Comment by kegsay** [comment]
So this has been in Labs for a while now, and isn't on a branch anymore. At the weekly discussion today we decided that this issue can be closed when:
 - EW looks at the `.well-known/matrix/client` response and pulls the proxy URL from it, enabling it by default if some `default: true` flag is set on the response. Otherwise, keep it toggleable in Labs.
 - Deploy the proxy to `:element.io` which then advertises the proxy so Element people can play with it very easily.

**Comment by Johennes** [comment]
Thanks @kegsay. I've updated the checklist in the description above. Do you rely on help from the app team for the remaining tasks?

**Comment by Johennes** [comment]
From talking to @andybalaam it sounds like the answer is _no_ for now.

**Comment by Johennes** [comment]
I'm removing this from the WAT project because empirically we haven't really needed to do much / anything out of the WAT to move this forward. Consequently, there's not much value in tracking it in our roadmap.

**Comment by kegsay** [comment]
This is mostly on me to deploy the SS proxy to `:element.io` and get EW looking at `.well-known`. The former is blocked on OIDC support which is being deployed to `:matrix.org` today. The latter isn't blocked on anything and could be done by anyone, but will probably be me.

**Comment by grinapo** [comment]
Is it possible to labs-enable it on released EW, suing self server/ssync proxy? Anything else needed to test apart from enabling?

**Comment by t3chguy** [comment]
Yes @grinapo via config - https://web-docs.element.dev/Element%20Web/config.html#labs-flags

**Comment by eternaleye** [comment]
Does the sliding sync proxy need to be deployed on element.io? Or would it suffice for the sliding sync proxy that is deployed on matrix.org already at `https://slidingsync.lab.matrix.org` (and which is listed in the `.well-known` entry) to add the necessary CORS headers permitting Element Web to make use of it?

**Comment by t3chguy** [comment]
@eternaleye the sliding proxy only connects to one server, so you cannot use https://slidingsync.lab.matrix.org/ for servers other than the one it connects to, which you can check using https://slidingsync.lab.matrix.org/client/server.json `"server": ["https://matrix-client.matrix.org"](https://matrix-client.matrix.org/),`

**Comment by EdGeraghty** [comment]
Hi, I wrote a patch to read the .well-known.

It's going to need a bit of TLC from someone who knows what they're doing, but this at least allows me to enable sliding sync from labs!

https://github.com/matrix-org/matrix-react-sdk/pull/12307

**Comment by dbkr** [comment]
Closing this in favour of a fresh one, https://github.com/element-hq/element-web/issues/28736 which tracks simplified sliding sync.
